### PR TITLE
Added sourcepoint tests to tcfv2 and ccpa

### PR DESCRIPTION
**Pull request by ioanna0**
- Adds sourcepoint unit tests to tcfv2 & ccpa

### CODE REVIEW

File: src/tcfv2/sourcepoint.test.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	});
+
+	it('should throw error if window scp exists', () => {
+		// eslint-disable-next-line no-underscore-dangle
+		window._sp_ = {};
+		expect(init).toThrow();
+	});
+
+	it("should initialize window _sp_ with the correct config if it doesn't exist", () => {
+		// eslint-disable-next-line no-underscore-dangle
```
Comment (sndrs) [inline src/tcfv2/sourcepoint.test.js:23]: this is handled by the one at the top
```suggestion
```

File: src/tcfv2/sourcepoint.test.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+jest.mock('../onConsentChange', () => ({
+	invokeCallbacks: jest.fn(),
+}));
+
+describe('Sourcepoint', () => {
+	afterEach(() => {
+		window._sp_ = undefined;
+	});
+
+	it('should throw error if window scp exists', () => {
```
Comment (sndrs) [inline src/tcfv2/sourcepoint.test.js:16]:
```suggestion
	it('should throw error if window._sp_ exists', () => {
```

File: src/tcfv2/sourcepoint.test.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		// eslint-disable-next-line no-underscore-dangle
+		window._sp_ = {};
+		expect(init).toThrow();
+	});
+
+	it("should initialize window _sp_ with the correct config if it doesn't exist", () => {
```
Comment (sndrs) [inline src/tcfv2/sourcepoint.test.js:22]:
```suggestion
	it("should initialize window._sp_ with the correct config if it doesn't exist", () => {
```

File: src/ccpa/sourcepoint.test.js [hunk 4]
```diff
@@ -1,15 +1,51 @@
...
+	invokeCallbacks: jest.fn(),
+}));
 
 describe('Sourcepoint', () => {
-	init();
+	afterEach(() => {
+		window._sp_ccpa = undefined;
+	});
+
+	it('should throw error if window scp exists', () => {
```
Comment (sndrs) [inline src/ccpa/sourcepoint.test.js:16]:
```suggestion
	it('should throw error if window._sp_ exists', () => {
```

File: src/ccpa/sourcepoint.test.js [hunk 5]
```diff
@@ -1,15 +1,51 @@
...
+		// eslint-disable-next-line no-underscore-dangle
```
Comment (sndrs) [inline src/ccpa/sourcepoint.test.js:17]:
```suggestion
```

File: src/ccpa/sourcepoint.test.js [hunk 6]
```diff
@@ -1,15 +1,51 @@
...
+		window._sp_ccpa = {};
+		expect(init).toThrow();
+	});
+
+	it("should initialize window ccpa with the correct config if it doesn't exist", () => {
+		// eslint-disable-next-line no-underscore-dangle
```
Comment (sndrs) [inline src/ccpa/sourcepoint.test.js:23]:
```suggestion
```

File: src/ccpa/sourcepoint.test.js [hunk 7]
```diff
@@ -1,15 +1,51 @@
...
+		window._sp_ccpa = undefined;
+	});
+
+	it('should throw error if window scp exists', () => {
+		// eslint-disable-next-line no-underscore-dangle
+		window._sp_ccpa = {};
+		expect(init).toThrow();
+	});
+
+	it("should initialize window ccpa with the correct config if it doesn't exist", () => {
```
Comment (sndrs) [inline src/ccpa/sourcepoint.test.js:22]:
```suggestion
	it("should initialize window._sp_ccpa_ with the correct config if it doesn't exist", () => {
```

File: src/ccpa/sourcepoint.test.js [hunk 8]
```diff
@@ -1,15 +1,51 @@
...
 import http from 'http';
 import url from 'url';
 import { init } from './sourcepoint';
+import { ACCOUNT_ID } from '../lib/accountId';
+
+jest.mock('../onConsentChange', () => ({
+	invokeCallbacks: jest.fn(),
+}));
 
 describe('Sourcepoint', () => {
```
Comment (sndrs) [inline src/ccpa/sourcepoint.test.js:11]:
```suggestion
describe('Sourcepoint CCPA', () => {
```

File: src/tcfv2/sourcepoint.test.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+import http from 'http';
+import url from 'url';
+import { init } from './sourcepoint';
+import { ACCOUNT_ID } from '../lib/accountId';
+
+jest.mock('../onConsentChange', () => ({
+	invokeCallbacks: jest.fn(),
+}));
+
+describe('Sourcepoint', () => {
```
Comment (sndrs) [inline src/tcfv2/sourcepoint.test.js:11]:
```suggestion
describe('Sourcepoint TCF', () => {
```

File: src/ccpa/sourcepoint.test.js (hunk 5 above)
Comment (ioanna0) [reply src/ccpa/sourcepoint.test.js:17]: oh wow vscode  auto save linter puts back automatically when i remove it 😳

File: src/index.test.js [hunk 10]
```diff
@@ -15,12 +15,12 @@ describe('cmp.init', () => {
 		expect(cmp.init).toThrow();
 	});
 
-	it('inititalises CCPA when in the US', () => {
+	it('initializes CCPA when in the US', () => {
 		cmp.init({ isInUsa: true });
```
Comment (mxdvl) [inline src/index.test.js:19]: Should the test not call it twice, to verify it only gets called once?

**Pull request merged**